configure.d/config_net_snmp_config_h

```diff
@@ -42,6 +42,12 @@ AH_BOTTOM([/* end of definitions added by configure on-the-fly */
 #define HAVE_AES 1
 #endif
 
+#ifdef NETSNMP_USE_OPENSSL
+/* Suppress deprecation warnings from OpenSSL 3.0+ */
+/* Our interface match OpenSSL 1.0.0 */
+#define OPENSSL_API_COMPAT 0x10000000L
```

Comment:
Will this definition end up in code that uses the Net-SNMP library? I'm wondering whether this definition should be moved into a private Net-SNMP header file.

Reply:
Yes, it will, as everything is including net-snmp-config.h, but I can't readily think of a possible candidate header for this?

Reply:
I don't think that it is acceptable to add this code in the public header file so it may be necessary to create a new header file.

Reply:
Any idea how to name it?

Reply:
How about `snmplib/openssl.h`? There are already several other private header files in that directory:
```
> ls snmplib/*h
snmplib/factory.h    snmplib/memcheck.h         snmplib/snmp_syslog.h
snmplib/inet_ntop.h  snmplib/snmpsm_init.h      snmplib/snprintf.h
snmplib/inet_pton.h  snmplib/snmpsm_shutdown.h  snmplib/ucd-snmp-includes.h
```

Reply:
Isn't the `include/` directory for public header files? Shouldn't private header files occur outside the `include/` directory?

Reply:
There are a number of headers there already which are not installed:
data_access/ openssl_md5.h agent/hardware/ agent/object_monitor.h library/openssl_???.h library/snmp_openssl.h 
so it is nmot unheard of :-)

Reply:
How about adding a new file in the include/net-snmp/library directory? It seems to me that the existing OpenSSL files in that directory come from the OpenSSL project so these files probably should not be modified. From commit 16fa55444e22af50be7eabbfbc71640d84285e7f: "local copies of the openssl headers".